[🐛] No signature of method: java.lang.Boolean.getStringValue() under android v14.1.0+

### What happened?

I've been using this library for a while (thank you!) and recently migrated to the expo config, which worked well under 14.0.1, then I upgraded to 14.2.1 and I've started getting the compilation under android failing with:
```
> Configure project :react-native-google-mobile-ads
:react-native-google-mobile-ads package.json found at /Projects/app/node_modules/react-native-google-mobile-ads/package.json
:app (dev) package.json found at /Projects/app/package.json

FAILURE: Build failed with an exception.

* Where:
Build file '/Projects/app/node_modules/react-native-google-mobile-ads/android/build.gradle' line: 85

* What went wrong:
A problem occurred evaluating project ':react-native-google-mobile-ads'.
> No signature of method: java.lang.Boolean.getStringValue() is applicable for argument types: (String, String) values: [android_app_id, ]

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

```

I've tried older versions and it also happened with 14.1.0 and 14.2.0. 14.0.1 is the latest that works, without doing any change in the config or code other than changing the version in the package.json. It both fails in local and EAS compilations

Ios is working (compiling through EAS, I cannot do it locally)

### Platforms

Only on Android

### React Native Info

```Shell
System:
  OS: Linux 6.10 Arch Linux
  CPU: (16) x64 AMD Ryzen 7 PRO 4750U with Radeon Graphics
  Memory: 8.60 GB / 29.10 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.14.0
    path: ~/.nodenv/versions/20.14.0/bin/node
  Yarn:
    version: 1.22.22
    path: /usr/bin/yarn
  npm:
    version: 10.7.0
    path: ~/.nodenv/versions/20.14.0/bin/npm
  Watchman:
    version: 20240414.112832.0
    path: /usr/bin/watchman
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: AI-241.18034.62.2411.12071903
Languages:
  Java:
    version: 17.0.12
    path: /usr/bin/javac
  Ruby:
    version: 3.3.2
    path: ~/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: ~18.2.0
  react-native:
    installed: 0.74.3
    wanted: 0.74.3
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```
```


### Are your using Typescript?

- [ ] My project is using Typescript

### package.json

```JSON
{
  ...
  "react-native-google-mobile-ads": "~14.0.1",
  ...
}
```
```


### app.json

```JSON
[
        'react-native-google-mobile-ads',
        {
          androidAppId: process.env.ADMOB_ANDROID_APP_ID,
          iosAppId: process.env.ADMOB_IOS_APP_ID,
          delayAppMeasurementInit: true,
          userTrackingUsageDescription: 'This identifier will be used to deliver personalized ads to you.',
          skAdNetworkItems: [
            'cstr6suwn9.skadnetwork',
            ...
          ]
```
```


### ios/Podfile

_No response_

### android/build.gradle

_No response_

### android/app/build.gradle

_No response_

### android/settings.gradle

_No response_

### AndroidManifest.xml

_No response_

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -74,29 +74,28 @@ project.ext {
 
 apply from: file("./app-json.gradle")
 
-def isManagedExpoProject = !rootProject.file("android").exists() && !rootProject.file("ios").exists()
 def appJSONGoogleMobileAdsAppIDString = ""
 def appJSONGoogleMobileAdsDelayAppMeasurementInitBool = false
 def appJSONGoogleMobileAdsOptimizeInitializationBool = true
 def appJSONGoogleMobileAdsOptimizeAdLoadingBool = true
 
-if (rootProject.ext.has("googleMobileAdsJson")) {
+if (rootProject.ext.has("googleMobileAdsJson") && rootProject.ext.googleMobileAdsJson) {
   appJSONGoogleMobileAdsAppIDString = rootProject.ext.googleMobileAdsJson.getStringValue("android_app_id", "")
   appJSONGoogleMobileAdsDelayAppMeasurementInitBool = rootProject.ext.googleMobileAdsJson.isFlagEnabled("delay_app_measurement_init", false)
   appJSONGoogleMobileAdsOptimizeInitializationBool = rootProject.ext.googleMobileAdsJson.isFlagEnabled("optimize_initialization", true)
   appJSONGoogleMobileAdsOptimizeAdLoadingBool = rootProject.ext.googleMobileAdsJson.isFlagEnabled("optimize_ad_loading", true)
 }
 
-if (!appJSONGoogleMobileAdsAppIDString && !isManagedExpoProject) {
+if (!appJSONGoogleMobileAdsAppIDString) {
   println "\n\n\n"
   println "**************************************************************************************************************"
   println "\n\n\n"
-  println "ERROR: react-native-google-mobile-ads requires an 'android_app_id' property inside a 'react-native-google-mobile-ads' key in your app.json."
-  println "  No android_app_id property was found in this location. The native Google Mobile Ads SDK will crash on startup without it."
+  println "WARNING: react-native-google-mobile-ads requires an 'android_app_id' property inside a 'react-native-google-mobile-ads' key in your app.json."
+  println "         No android_app_id property was found in this location. The native Google Mobile Ads SDK will crash on startup without it."
+  println "         You can safely ignore this warning if you are using our Expo config plugin."
   println "\n\n\n"
   println "**************************************************************************************************************"
   println "\n\n\n"
-  System.exit(1)
 }
 
 android {

--- a/ios_config.sh
+++ b/ios_config.sh
@@ -88,12 +88,6 @@ while true; do
   _CURRENT_LOOKUPS=$((_CURRENT_LOOKUPS+1))
 done
 
-# Bail out if project is a managed Expo project:
-if [[ ! -d "${PROJECT_DIR}/ios" ]] && [[ ! -d "${PROJECT_DIR}/android" ]]; then
-  echo "info: Project does not contain an ios or android folder, assume it's a managed Expo project using our Expo Config Plugin."
-  exit 0
-fi
-
 if [[ ${_SEARCH_RESULT} ]]; then
   _JSON_OUTPUT_RAW=$(cat "${_SEARCH_RESULT}")
   _RN_ROOT_EXISTS=$(ruby -KU -e "require 'rubygems';require 'json'; output=JSON.parse('$_JSON_OUTPUT_RAW'); puts output[$_JSON_ROOT]" || echo '')
@@ -169,8 +163,9 @@ if ! [[ -f "${_TARGET_PLIST}" ]]; then
 fi
 
 if ! [[ $_IOS_APP_ID ]]; then
-  echo "error: ios_app_id key not found in react-native-google-mobile-ads key in app.json. App will crash without it."
-  exit 1
+  echo "warning: ios_app_id key not found in react-native-google-mobile-ads key in app.json. App will crash without it."
+  echo "         You can safely ignore this warning if you are using our Expo config plugin."
+  exit 0
 fi
 
 for plist in "${_TARGET_PLIST}" "${_DSYM_PLIST}" ; do

